Re-enable hoauth2 and yesod-auth-oauth2

pbrisbin opened the pull request:
[hoauth2-2.8.0][hoauth2] seems to have corrected the upper bounds on
memory and text that caused this package to be disabled. It was blocking
our package, yesod-auth-oauth2 which compiles fine with the following
stack.yaml:

```yaml
resolver: nightly-2023-04-05
extra-deps:
  - hoauth2-2.8.0
```

[hoauth2]: https://hackage.haskell.org/package/hoauth2-2.8.0

**Side-note**: how is a situation like this meant to be resolved? If I hadn't noticed and investigated, yesod-auth-oauth2 would've probably dropped out in the next LTS, which is a bigger deal to get corrected (I think). I try to stay on top of these things, but I can't keep track of what automation exists. For example, once hoauth2-2.8.0 came out, I (naively) thought some process would check to see if it worked now. I've also lost track of how/when maintainers are supposed to be notified. I searched and couldn't find any Issue on this or my repository about when this happened originally. I see [this one](https://github.com/commercialhaskell/stackage/issues/6486), but that says 2021-12-30 and we're still present in 2022-11-17, so I think it's unrelated. Is there any documentation I'm missing?

pbrisbin commented:
CI failure seems unrelated to me. Please let me know if I should change anything in response to it.

juhp commented:
Yea, it was disabled by commit 94a341a6 when nightly was updated to 9.4.2.
I can certainly recommend checking for your packages after Nightly gets rebased.
Unfortunately stackage is still largely a semi-manual process, though we have
an offline cli tool that can help with rechecking disabled packages sometimes.
We haven't always been consistent with notifying maintainers of disabled packages, sorry:
It also doesn't scale that well...

I personally try to check for dependent packages when I re-enable a package,
but probably not all the curators do that.

I also encourage people to build their projects (libraries) across Stackage major versions
eg using stack-all or CI.

pbrisbin commented:
Thanks very much for the response @juhp.

> I can certainly recommend checking for your packages after Nightly gets rebased

Just curious, is there any way to subscribe to be notified of this event?

> Unfortunately stackage is still largely a semi-manual process, though we have
an offline cli tool that can help with rechecking disabled packages, which we use sometimes.
We haven't always been consistent with notifying maintainers of disabled packages, sorry:
that also doesn't scale that well I think

Yeah, understood. Just to be totally clear, I was not complaining or requesting some automation; rather, I was just making sure I had not missed something. It seems the process just has (understandble) gaps, and being manually on top of the situation is the best we can all collectively do. Thanks for your efforts!

> I also encourage people to build their projects (libraries) across Stackage major versions
eg using stack-all or CI

Yes, we are very much on top of this and have a full matrix, including nightly, on all our projects. I suspect that what happened was we saw our nightly case fail in CI due to `hoauth2`. Since this was not caused by our project, we added the `extra-dep` and moved on. In retrospect, we should've tracked something on our end at that point to look out for the updated `hoauth2`.
